prov/shm/src/smr_init.c

```diff
@@ -94,20 +109,19 @@ static int smr_getinfo(uint32_t version, const char *node, const char *service,
 	struct fi_info *cur;
 	uint64_t mr_mode, msg_order;
 	int fast_rma;
-	int ptrace_scope, ret;
+	int ret;
 
 	mr_mode = hints && hints->domain_attr ? hints->domain_attr->mr_mode :
 						FI_MR_VIRT_ADDR;
 	msg_order = hints && hints->tx_attr ? hints->tx_attr->msg_order : 0;
+	smr_check_ptrace_scope();
```

Comment:
I doubt this will change between getinfo calls. Can this be moved to a one-time initialization routine?  It can be separate from SHM_INI.  Adding a static bool to the check call might work as well.

Reply:
Updated